src/Lexer.php

```diff
@@ -42,11 +53,17 @@ class Lexer
     private int $lineNumber;
     private bool $eos;
     /**
-     * @var array<string, string>
+     * @phpstan-var array<TKeywordsType, TKeywordsString>
      */
     private array $keywordsCache = [];
     /**
-     * @var array<string, list<string>>
+     * @var array{}|array{
+     *     Given: list<string>,
+     *     When: list<string>,
+     *     Then: list<string>,
+     *     And: list<string>,
+     *     But: list<string>,
```

Comment:
From the code I understood that this array should have that exact structure, which is why I opted to go for a strict approach instead.